src/rdkafka_admin.c

```diff
@@ -1750,15 +1750,14 @@ rd_kafka_CreatePartitionsResponse_parse (rd_kafka_op_t *rko_req,
                                 errstr = (char *)rd_kafka_err2str(error_code);
                         else
                                 RD_KAFKAP_STR_DUPA(&errstr, &error_msg);
+                } else {
+                        errstr = NULL;
```

Comment:
errstr is the parent's buffer, which might be needed later in this function, so suggest not setting it to NULL here.
Instead change the result_new(..) call below to `error_code ? errstr : NULL`